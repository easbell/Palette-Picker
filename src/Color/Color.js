import React from 'react';

export const Color = ({color}) => {

  return (
    <div style={{backgroundColor: color}} className='color'>
      {color}
    </div>
  )
}

export default Color