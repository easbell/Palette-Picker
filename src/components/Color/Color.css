.color {
  width: 20%;
  display: flex;
  align-items: center;
  justify-content: center;
}