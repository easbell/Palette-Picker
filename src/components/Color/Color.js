import React, { Component } from 'react';
import './Color.css';
import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
import Tooltip from 'react-bootstrap/Tooltip';
import Button from 'react-bootstrap/Button';

export class Color extends Component {
  constructor() {
    super();
    this.state = {
      locked: false
    }
  }

  toggleLock = () => {
    const { lockColor, color } = this.props;
    this.setState({ locked: !this.state.locked })
    lockColor(color)
  }

  render() {
    const { color } = this.props;
    const { locked } = this.state;
    return (
      <div style={{backgroundColor: color}} onClick={this.toggleLock} className='color'>
        <OverlayTrigger
          placement={'top'}
          overlay={
            <Tooltip id={`tooltip-${'top'}`}>
              Click to lock color.
            </Tooltip>
          }>
        <Button variant="secondary" className='lock'>
          { locked 
            ? <i className="fas fa-lock"></i>
            : <i className="fas fa-lock-open"></i>
          }
        </Button>
        </OverlayTrigger>
      </div>
    )
  }
}

export default Color;