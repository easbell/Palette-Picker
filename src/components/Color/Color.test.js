import React from 'react';
import { Color } from './Color';
import { shallow } from 'enzyme';

describe('Color', () => {
  let wrapper;
  let mockFn;
  beforeEach(() => {
    mockFn = jest.fn();

    wrapper = shallow(
      <Color 
        lockColor={mockFn} 
        color={'#ffffff'} 
        savePalette={mockFn}
      />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state and call lockColor when color is clicked', () => {
    wrapper.find('.color').simulate('click');
    expect(wrapper.state('locked')).toBe(true);
    expect(mockFn).toHaveBeenCalled();
  });
});