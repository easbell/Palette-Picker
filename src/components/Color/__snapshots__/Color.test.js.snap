// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color should match snapshot 1`] = `
<div
  className="color"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <OverlayTrigger
    defaultOverlayShown={false}
    overlay={
      <ForwardRef(render)
        id="tooltip-top"
      >
        Click to lock color.
      </ForwardRef(render)>
    }
    placement="top"
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <Button
      active={false}
      className="lock"
      disabled={false}
      type="button"
      variant="secondary"
    >
      <i
        className="fas fa-lock-open"
      />
    </Button>
  </OverlayTrigger>
</div>
`;
